src/components/node-graph/components/node-slot.tsx

```diff
@@ -1,44 +1,168 @@
 import { Handle, HandleType, Position as HandlePosition, useNodeConnections } from "@xyflow/react";
-import { Trash } from "lucide-react";
+import { ArrowBigRightIcon, PlusIcon, TrashIcon } from "lucide-react";
 import { twJoin } from "tailwind-merge";
+import { useDebouncedCallback } from "use-debounce";
 
 import { StepSocket } from "@/api";
+import NodeInput from "@/components/node-graph/components/step/node-input";
+import { Badge } from "@/components/ui/badge";
 import { Button } from "@/components/ui/button";
-import { cn } from "@/lib/utils";
-
-import NodeInput from "./step/node-input";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { Popover, PopoverContent, PopoverTrigger } from "@/components/ui/popover";
+import { cn, isFile } from "@/lib/utils";
 
 interface NodeSlotProps {
-  socket: StepSocket;
   type: HandleType;
+  socket: StepSocket;
+  // Edit props
+  onEditLabel?: (newSocketLabel: string) => void;
+  onEditData?: (newSocketData: string | boolean | number) => void;
+  onDelete?: () => void;
+  // Styling props
   hideLabel?: boolean;
-  edit?: boolean;
-  allowEditSockets?: boolean;
-  onEditSocketLabel?: (newSocketLabel: string) => void;
-  onDeleteSocket?: () => void;
-  style?: React.CSSProperties;
+  handleStyle?: React.CSSProperties;
 }
 
-export function NodeSlot({
+const DataSocketDefaultValuePopover = ({
+  children,
+  onValueChanged,
+}: {
+  children?: React.ReactNode;
+  onValueChanged?: (newSocketData: string | boolean | number) => void;
+}) => {
+  const debouncedOnValueChanged = useDebouncedCallback((value: string) => {
+    let parsedValue: string | boolean | number = value;
+
+    if (value.startsWith('"') && value.endsWith('"')) parsedValue = value.slice(1, -1);
+    else if (value.toLowerCase() === "true") parsedValue = true;
+    else if (value.toLowerCase() === "false") parsedValue = false;
+    else if (!isNaN(Number(value))) parsedValue = Number(value);
+
+    if (onValueChanged) onValueChanged(parsedValue);
+  }, 500);
```

Comment:
could you help make node-input also use this so that it can work again? currently i cant edit node inputs without selecting the whole thing to replace 👀

Reply:
will fix this issue in another PR, it was not as simple as i thought it would be since `<NodeInput>` is used for both viewing and editing